static/js/overlays.js

```diff
@@ -279,6 +281,16 @@ export function close_modal(selector, conf) {
     blueslip.debug("close modal: " + selector);
 
     if (conf && conf.micromodal) {
+        const id_selector = `#${selector}`;
+        const micromodal = $(id_selector);
+
+        micromodal.find(".modal__container").on("animationend", (event) => {
+            const animation_name = event.originalEvent.animationName;
+            if (animation_name === "mmfadeOut" && conf.on_hidden) {
+                conf.on_hidden();
```

Comment:
Is this approach documented by Micromodal upstream?  

Also does https://stackoverflow.com/questions/55560506/how-do-i-have-a-custom-callback-upon-close-using-micromodal not work for this purpose?

Reply:
(I just want to make sure we have an understanding of whether this might break when upgrading Micromodal; if they don't have a documented way to do this, but this works, we can open an issue with them requesting one, and then add a comment here linking to that issue, and plan to replace this code if they do a future release adding one)

Reply:
> Is this approach documented by Micromodal upstream?

No.
> Also does https://stackoverflow.com/questions/55560506/how-do-i-have-a-custom-callback-upon-close-using-micromodal not work for this purpose?

`onClose` is for when the modal is triggered to close. We make use of it here:
https://github.com/zulip/zulip/blob/410281624098f56ff9a67c8c17aca205487d59bf/static/js/overlays.js#L194-L198

Reply:
> I just want to make sure we have an understanding of whether this might break when upgrading Micromodal

This approach depends on the CSS animation we use, which is our own. It is independent of Micromodal.
https://github.com/zulip/zulip/blob/410281624098f56ff9a67c8c17aca205487d59bf/static/styles/modal.css#L180-L194

Reply:
OK, I can merge this after adding some comments.  Let's open an issue with Micromodal upstream asking for an `afterClose` hook that gets called once the close process is complete?

Reply:
Actually, no, I'm confused -- it looks like we already have this feature!  There's this existing block of code in the `open_modal` code path, which I think registers these handlers properly:

```
        micromodal.find(".modal__container").on("animationend", (event) => {                                                    
            // Micromodal doesn't trigger an event when a modal                                                                 
            // finishes being hidden (`hidden.bs.modal` in Bootstrap).                                                          
            // We work around this by listening for when the                                                                    
            // animations we've configured when a Micromodal is closed                                                          
            // complete.                                                                                                        
            const animation_name = event.originalEvent.animationName;                                                           
            if (animation_name === "mmfadeIn") {                                                                                
                // Equivalent to bootstrap's "shown.bs.modal" event                                                             
                                                                                                                                
                // Micromodal adds the is-open class before the modal                                                           
                // animation is complete, which isn't really helpful                                                            
                // since a modal is only fully open after the                                                                   
                // animation is complete. So, we manually add a class                                                           
                // after the animation is complete.                                                                             
                micromodal.addClass("modal--open");                                                                             
                micromodal.removeClass("modal--opening");                                                                       
                                                                                                                                
                if (conf.on_shown) {                                                                                            
                    conf.on_shown();                                                                                            
                }                                                                                                               
            } else if (animation_name === "mmfadeOut") {                                                                        
                // Equivalent to bootstrap's "hidden.bs.modal" event                                                            
                                                                                                                                
                micromodal.removeClass("modal--open");                                                                          
                if (conf.autoremove) {                                                                                          
                    micromodal.remove();                                                                                        
                }                                                                                                               
                if (conf.on_hidden) {                                                                                           
                    conf.on_hidden();                                                                                           
                }                                                                                                               
            }                                                                                                                   
        });                                                                                                                     
```

Reply:
> OK, I can merge this after adding some comments. Let's open an issue with Micromodal upstream asking for an `afterClose` hook that gets called once the close process is complete?

That hook depends on the CSS (animation), which is our own. Micromodal just handles the JS part of the modal. So, I don't think they would add such a hook

Reply:
> Actually, no, I'm confused -- it looks like we already have this feature! There's this existing block of code in the open_modal code path, which I think registers these handlers properly

That code path attaches the handlers before opening the modal. This PR adds that functionality to when closing a modal too. This is helpful in cases where we want to attach the handler only if the modal is closed in a certain way, e.g. as in #20373, where the handler should run only if the modal is closed by clicking on the "Deactivate" button.

Reply:
Hmm, but can't we just add an unconditional on-close handler function, where that function contains an `if` statement that checks some state within the application code module?  

@sahil839 for thoughts.

Reply:
I guess I can see this being a convenience; added appropriate documentation and will merge.